influxql/ast.go

```diff
@@ -1036,27 +1036,76 @@ func (s *SelectStatement) validateAggregates(tr targetRequirement) error {
 			numAggregates++
 		}
 	}
-	if numAggregates != 0 && numAggregates != len(s.Fields) {
-		return fmt.Errorf("mixing aggregate and non-aggregate queries is not supported")
+
+	allowMixedAggregates := func() error {
+		if numAggregates != 0 && numAggregates != len(s.Fields) {
+			return fmt.Errorf("mixing aggregate and non-aggregate queries is not supported")
+		}
+		return nil
 	}
 
-	// Secondly, determine if specific calls have at least one and only one argument
+	// Secondly, determine if specific calls have the correct number of arguments
```

Comment:
This still needs more refactoring.  I wanted to at least get it all in the same spot for now.